The target "CreateManifestResourceNames" does not exist in the project.

I get this error when I try to build this project:

> The target "CreateManifestResourceNames" does not exist in the project.	


```xml

<Project Sdk="Microsoft.Build.Traversal/2.0.12"  ToolsVersion="15.0">
  <ItemGroup>
    <!-- Build all projects recursively under the "src" folder -->
    <ProjectReference Include="src\**\*.*proj" />
  </ItemGroup>
</Project>


```

Hitting the same with msbuild 16.6 .NET Core in dotnet/runtime:

```
C:\git\runtime4\.dotnet\sdk\5.0.100-preview.3.20170.6\Microsoft.Common.CurrentVersion.targets(2795,7): error MSB4057: The target "CreateManifestResourceNames" does not exist in the project. [C:\git\runtime4\src\libraries\Native\native-binplace.proj]
```

I've just met this issue and found it to occur when _not_ using `.csproj`

Failing:
```
  Restore completed in 35.68 ms for ...\packages\project.proj.
Microsoft (R) Build Engine version 16.5.0+d4cbfca49 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 34.2 ms for ...\packages\project.proj.
C:\Program Files\dotnet\sdk\3.1.201\Microsoft.Common.CurrentVersion.targets(2795,7): error MSB4057: The target "CreateManifestResourceNames" does not exist in the project. [...\packages\project.proj]
```

Success:
```
  Restore completed in 2.71 sec for ...\packages\project.csproj.
Microsoft (R) Build Engine version 16.5.0+d4cbfca49 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 41.49 ms for ...\packages\project.csproj.
  project -> ...\packages\bin\Debug\net452\project.dll
  project -> ...\packages\results\
```

--- a/Packages.props
+++ b/Packages.props
@@ -1,10 +1,10 @@
 <Project>
   <ItemGroup>
-    <PackageReference Update="Microsoft.Build" Version="16.6.0" />
-    <PackageReference Update="Microsoft.Build.Framework" Version="16.6.0" />
-    <PackageReference Update="Microsoft.Build.Utilities.Core" Version="16.6.0" />
+    <PackageReference Update="Microsoft.Build" Version="16.8.0" />
+    <PackageReference Update="Microsoft.Build.Framework" Version="16.8.0" />
+    <PackageReference Update="Microsoft.Build.Utilities.Core" Version="16.8.0" />
     <PackageReference Update="Microsoft.NET.Test.Sdk" Version="16.7.1" />
-    <PackageReference Update="MSBuild.ProjectCreation" Version="1.4.6" />
+    <PackageReference Update="MSBuild.ProjectCreation" Version="2.1.1" />
     <PackageReference Update="Shouldly" Version="3.0.2" />
     <PackageReference Update="xunit" Version="2.4.1" />
     <PackageReference Update="xunit.runner.visualstudio" Version="2.4.3" />

--- a/src/NoTargets.UnitTests/NoTargetsTests.cs
+++ b/src/NoTargets.UnitTests/NoTargetsTests.cs
@@ -220,6 +220,35 @@ namespace Microsoft.Build.NoTargets.UnitTests
         [Theory]
         [InlineData(".csproj")]
         [InlineData(".proj")]
+        public void PublishWithNoBuild(string projectExtension)
+        {
+            var f = ProjectCreator.Templates.NoTargetsProject(
+                    path: GetTempFileWithExtension(projectExtension),
+                    targetFramework: "netcoreapp3.1",
+                    customAction: creator =>
+                    {
+                        creator
+                            .Property("RuntimeIdentifier", "win-x64")
+                            .Property("Platforms", "x64")
+                            .Target("TakeAction", afterTargets: "Build")
+                                .TaskMessage("2EA26E6FC5C842B682AA26096A769E07", MessageImportance.High);
+                    })
+                .Save()
+                .TryBuild(restore: true, "Build", out bool buildResult, out BuildOutput buildOutput)
+                .TryBuild("Publish", new Dictionary<string, string> { ["NoBuild"] = "true" }, out bool publishResult, out BuildOutput publishOutput);
+
+            buildResult.ShouldBeTrue(() => buildOutput.GetConsoleLog());
+
+            buildOutput.Messages.High.ShouldContain("2EA26E6FC5C842B682AA26096A769E07");
+
+            publishResult.ShouldBeTrue(() => publishOutput.GetConsoleLog());
+
+            publishOutput.Messages.High.ShouldNotContain("2EA26E6FC5C842B682AA26096A769E07");
+        }
+
+        [Theory]
+        [InlineData(".csproj")]
+        [InlineData(".proj")]
         public void SimpleBuild(string projectExtension)
         {
             ProjectCreator.Templates.NoTargetsProject(

--- a/src/NoTargets/Sdk/Sdk.props
+++ b/src/NoTargets/Sdk/Sdk.props
@@ -32,6 +32,13 @@
 
     <!-- Targeting packs shouldn't be referenced as traversal projects don't compile. -->
     <DisableImplicitFrameworkReferences Condition="'$(DisableImplicitFrameworkReferences)' == ''">true</DisableImplicitFrameworkReferences>
+
+    <!-- Disable publish actions -->
+    <CopyBuildOutputToPublishDirectory Condition="'$(CopyBuildOutputToPublishDirectory)' == ''">false</CopyBuildOutputToPublishDirectory>
+    <CopyOutputSymbolsToPublishDirectory Condition="'$(CopyOutputSymbolsToPublishDirectory)' == ''">false</CopyOutputSymbolsToPublishDirectory>
+
+    <!-- Don't generate a deps file -->
+    <GenerateDependencyFile Condition="'$(GenerateDependencyFile)' == ''">false</GenerateDependencyFile>
   </PropertyGroup>
 
   <ItemDefinitionGroup Condition=" '$(NoTargetsDoNotReferenceOutputAssemblies)' != 'false' ">
@@ -45,11 +52,12 @@
     </ProjectReference>
   </ItemDefinitionGroup>
 
+  <Target Name="CreateManifestResourceNames" />
+
   <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" Condition=" '$(MicrosoftCommonPropsHasBeenImported)' != 'true' "/>
 
   <Import Project="$(CustomAfterNoTargetsProps)" Condition=" '$(CustomAfterNoTargetsProps)' != '' And Exists('$(CustomAfterNoTargetsProps)') " />
 
   <!-- For CPS/VS support. Importing in .props allows any subsequent targets to redefine this if needed -->
   <Target Name="CompileDesignTime" />
-  
 </Project>

